### ISSUE TITLE
BDN 11.3: Any way to exclude mandatory validators?

### ISSUE THREAD

**Issue by ig-sinicyn** [dotnet/BenchmarkDotNet#988]
As far as I can see, logic in the `GetFullConfig()` is overriden by the implementation of the `CompositeValidator` ctor:
```cs
        public CompositeValidator(IValidator[] configuredValidators)
        {
            Validators = configuredValidators
                .Concat(MandatoryValidators)
                .GroupBy(validator => validator.GetType())
                .Select(groupedByType => groupedByType.FirstOrDefault(validator => validator.TreatsWarningsAsErrors) ?? groupedByType.First())
                .Distinct()
                .ToArray();
        }
```
Can this behavior be reverted?

I have custom validators, my integration tests have an assertion that checks there's no validation errors and the JitOptimizationsValidator breaks my tests in debug mode.

As a workaround I've made advanced assertion that ignores warnings from mandatory validators but I'd prefer to have full control over benchmark config.

**Comment by AndreyAkinshin** [comment]
@adamsitnik what do you think?

**Comment by ig-sinicyn** [comment]
I'd suggest to threat the `BenchmarkRunInfo` as a class that contains final benchmark description.

Resulting behavior will be:
* All public API overloads that do not accept the BenchmarkRunInfo may alter the config.
* All public API overloads that accept BenchmarkRunInfo use the config as is.

**Comment by adamsitnik** [comment]
I think that over the time, we added too many configuration options. Our code gets more and more complex, while some of the features are used by single users.

Having said that, I think that it's not worth the gain to add the possibility to disable the mandatory validators.

@ig-sinicyn I really appreciate your help and input, I hope that you will understand.

**Comment by ig-sinicyn** [comment]
Ok, got it:)

**Comment by adamsitnik** [comment]
btw @ig-sinicyn I think that the lesser evil would be for you to use reflection and simply set [this field](https://github.com/dotnet/BenchmarkDotNet/blob/cebe2a0f84fa21acb6db9613fe3a4326d635f129/src/BenchmarkDotNet/Validators/CompositeValidator.cs#L8) to an empty array

**Comment by adamsitnik** [comment]
@ig-sinicyn I finally got some time to finish my big Config refactor (see #912). Now we create config once, when we have all information and it's final = immutable.

I also added the feature that you have implemented in #1027

```cs
var config = config.With(ConfigOptions.DisableOptimizationsValidator);
```

I introduced the concept of config.Options which is just an enum flag. So we can have up to 31/63 boolean flags in config without changing the interface.
